src/main/java/controllers/BookstoreRestController.java

```diff
@@ -0,0 +1,202 @@
+package controllers;
+
+import models.*;
+import repositories.*;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+
+@RestController
+public class BookstoreRestController {
+
+    @Autowired
+    private BookRepository bookRepository;
+    @Autowired
+    private BookstoreRepository bookstoreRepository;
+    @Autowired
+    private BookstoreOwnerRepository bookstoreOwnerRepository;
+    @Autowired
+    private CustomerRepository customerRepository;
+    @Autowired
+    private ShoppingCartRepository shoppingCartRepository;
+    @Autowired
+    private SaleRepository saleRepository;
+
+
+    //BookstoreOwner REST endpoints
+    @GetMapping("/api/getBookstoreOwner")
+    public BookstoreOwner getBookstoreOwner(@RequestParam(value = "id") long id) {
+        return bookstoreOwnerRepository.findById(id);
+    }
+
+    @GetMapping("/api/getBookstoreOwners")
+    public Iterable<BookstoreOwner> getBookstoreOwner() {
+        return bookstoreOwnerRepository.findAll();
+    }
+
+    @PostMapping("/api/newBookstoreOwner")
+    public BookstoreOwner newBookstoreOwner(@RequestParam(value = "name") String name) {
+        BookstoreOwner bookstoreOwner = new BookstoreOwner(name);
+        bookstoreOwnerRepository.save(bookstoreOwner);
+        return bookstoreOwner;
+    }
+
+
+    //Bookstore REST endpoints
+    @GetMapping("/api/getBookstore")
+    public Bookstore getBookstore(@RequestParam(value = "id") long id) {
+        return bookstoreRepository.findById(id);
+    }
+
+    @GetMapping("/api/getBookstores")
+    public Iterable<Bookstore> getBookstores() {
+        return bookstoreRepository.findAll();
+    }
+
+    @PostMapping("/api/newBookstore")
+    public Bookstore newBookstore(@RequestParam(value = "name") String name, @RequestParam(value = "bookstoreOwnerID") long bookstoreOwnerID) {
+        BookstoreOwner bookstoreOwner = bookstoreOwnerRepository.findById(bookstoreOwnerID);
+        if (bookstoreOwner == null)
+            return null;
+        Bookstore bookstore = new Bookstore(name);
+        bookstore.setBookstoreOwner(bookstoreOwner);
+        bookstoreOwner.addBookstore(bookstore);
+        bookstoreRepository.save(bookstore);
+        bookstoreOwnerRepository.save(bookstoreOwner);
+        return bookstore;
+    }
+
+
+    //Book REST endpoints
+    @GetMapping("/api/getBook")
+    public Book getBook(@RequestParam(value = "id") long id) {
```

Comment:
can we change name from id to bookId, this can potentially be confusing/ambiguous on the client's side

Reply:
Addressed in f06ac09992047c1a3eab5bea5cc128da3024d6f4